bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/discovery/IRobotDiscoveryService.java

```diff
@@ -0,0 +1,167 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.irobot.discovery;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.json.JSONException;
+import org.openhab.binding.irobot.IRobotBindingConstants;
+import org.openhab.binding.irobot.internal.IdentProtocol;
+import org.openhab.binding.irobot.internal.IdentProtocol.IdentData;
+import org.openhab.binding.irobot.roomba.RoombaConfiguration;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.net.NetUtil;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Discovery service for iRobot's
+ *
+ * @author Pavel Fedin <pavel_fedin@mail.ru> - Initial contribution
+ *
+ */
+@Component(service = DiscoveryService.class)
+@NonNullByDefault
+public class IRobotDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(IRobotDiscoveryService.class);
+    private final Runnable scanner;
+    private @Nullable ScheduledFuture<?> backgroundFuture;
+
+    public IRobotDiscoveryService() {
+        super(Collections.singleton(IRobotBindingConstants.THING_TYPE_ROOMBA), 30, true);
+        scanner = createScanner();
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.trace("Starting background discovery");
+        stopBackgroundScan();
+        backgroundFuture = scheduler.scheduleWithFixedDelay(scanner, 0, 60, TimeUnit.SECONDS);
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        logger.trace("Stopping background discovery");
+        stopBackgroundScan();
+        super.stopBackgroundDiscovery();
+    }
+
+    private void stopBackgroundScan() {
+        ScheduledFuture<?> scan = backgroundFuture;
+
+        if (scan != null && !scan.isDone()) {
+            scan.cancel(true);
+            backgroundFuture = null;
+        }
+    }
+
+    @Override
+    protected void startScan() {
+        logger.trace("startScan");
```

Comment:
Yes, i remember this. To tell the truth i don't know framework's own logging facilities, and there could be tons of other output there, so for debugging it's just easier to have this here. And this line does not produce a lot of output, it's triggered only on startup. Also it helps to make sure that everything is OK on user's side, you can simply ask to crank up your binding logs to TRACE.
 But, OK, i'll remove it because this is the requirement.